code/datums/components/footstep.dm

```diff
@@ -0,0 +1,133 @@
+///Footstep component. Plays footsteps at parents location when it is appropriate.
+/datum/component/footstep
+	///How many steps the parent has taken since the last time a footstep was played.
+	var/steps = 0
+	///volume determines the extra volume of the footstep. This is multiplied by the base volume, should there be one.
+	var/volume
+	///e_range stands for extra range - aka how far the sound can be heard. This is added to the base value and ignored if there isn't a base value.
+	var/e_range
+	///footstep_type is a define which determines what kind of sounds should get chosen.
+	var/footstep_type
+	///This can be a list OR a soundfile OR null. Determines whatever sound gets played.
+	var/footstep_sounds
+	///Whether or not to add variation to the sounds played
+	var/sound_vary = FALSE
+
+/datum/component/footstep/Initialize(footstep_type_ = FOOTSTEP_MOB_BAREFOOT, volume_ = 0.5, e_range_ = -8, vary)
+	if(!ismovable(parent))
+		return COMPONENT_INCOMPATIBLE
+	volume = volume_
+	e_range = e_range_
+	footstep_type = footstep_type_
+	sound_vary = vary
+	switch(footstep_type)
+		if(FOOTSTEP_MOB_HUMAN)
+			if(!ishuman(parent))
+				return COMPONENT_INCOMPATIBLE
+			RegisterSignal(parent, list(COMSIG_MOVABLE_MOVED), .proc/play_humanstep)
```

Comment:
There's also 3 other unnecessary `list()`s inside of the `RegisterSignal()`s below, but I won't make a review for each. 
```suggestion
			RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/play_humanstep)
```